### PR TITLE
Flythrough fix

### PR DESCRIPTION
**Pull request by DamianMadden**
Resolves [AB#606](https://dev.azure.com/euclideon/57c3bd2a-8f94-4578-b195-2e4fa9d8f295/_workitems/edit/606)

Well I never observed the crash described but there were errors and this probably fixes the circumstances in which it might crash

### CODE REVIEW

File: src/scene/vcPOI.cpp [hunk 1]
```diff
@@ -145,15 +145,28 @@ bool vcPOI::GetPointAtDistanceAlongLine(double distance, udDouble3 *pPoint, int
...
-    if (totalDist + segmentLength > distance)
+
+    if (!m_line.closed && seg0 == m_line.numPoints - 1)
+    {
+      if (pPoint != nullptr)
+        *pPoint = m_line.pPoints[seg0];
+
+      if (pSegmentIndex != nullptr)
+        *pSegmentIndex = seg0;
+
```
[the remark this answers is not in the document]
Comment (DamianMadden) [reply src/scene/vcPOI.cpp]: if line !closed, yes

[the remark this answers is not in the document]
Comment (DamianMadden) [reply src/scene/vcPOI.cpp]: wasn't that how flythroughs worked?

File: src/scene/vcPOI.cpp [hunk 2]
```diff
@@ -145,15 +145,28 @@ bool vcPOI::GetPointAtDistanceAlongLine(double distance, udDouble3 *pPoint, int
...
+      if (pSegmentProgress != nullptr)
+        *pSegmentProgress = 0.0;
+
+      return true;
```
[the remark this answers is not in the document]
Comment (DamianMadden) [reply src/scene/vcPOI.cpp]: wouldn't based on your logic so far any input distance > 0 into this function return a non-zero change in position that could be used in orientation calculations?

File: src/scene/vcPOI.cpp [hunk 3]
```diff
@@ -288,9 +301,15 @@ void vcPOI::AddToScene(vcState *pProgramState, vcRenderData *pRenderData)
...
+      {
         pProgramState->camera.eulerRotation = udDirectionToYPR(m_line.pPoints[1] - m_line.pPoints[0]);
+      }
       else
-        pProgramState->camera.eulerRotation = udSlerp(udDoubleQuat::create(startYPR), udDoubleQuat::create(udDirectionToYPR(updatedPosition - pProgramState->camera.position)), 0.2).eulerAngles();
+      {
+        udDouble3 direction = updatedPosition - pProgramState->camera.position;
+        udDouble3 endYPR = udDirectionToYPR(direction == udDouble3::zero() ? m_line.pPoints[m_line.numPoints - 1] - m_line.pPoints[m_line.numPoints - 2] : direction);
+        pProgramState->camera.eulerRotation = udSlerp(udDoubleQuat::create(startYPR), udDoubleQuat::create(endYPR), 0.2).eulerAngles();
+      }
```
[the remark this answers is not in the document]
Comment (DamianMadden) [reply src/scene/vcPOI.cpp]: if I don't change the return type stuff this is either a vector between the first 2 points or triggers an error at the end position when updatedPosition == camera position

**Comment by OsiJr** [comment]
Was closed accidentally because target branch was merged and deleted.

**Comment by OsiJr** [comment]
Closing this as the bug it fixes cannot be reproduced.

**Pull request closed without merging**